packages/sdk/src/deserialization.ts

```diff
@@ -79,6 +81,14 @@ export function deserializeAccountTransaction(serializedTransaction: Cursor): {
     const accountTransactionHandler = getAccountTransactionHandler(transactionType);
     const payload = accountTransactionHandler.deserialize(serializedTransaction);
 
+    if (transactionType == AccountTransactionType.InitContract) {
+        console.log('Initcontract transaction type detected');
+        if (header.energyAmount !== undefined) {
+            console.log('InitContractPayload detected, setting energy in payload', header.energyAmount);
+            (payload as InitContractPayload).maxContractExecutionEnergy = header.energyAmount;
+        }
+    }
```

Comment:
I also think the better option here would be to do breaking changes into a new major release and fix this at the type level where `maxContractExecutionEnergy` is incorrectly defined..